Dockerfile

```
@@ -0,0 +1,27 @@
+FROM ubuntu:xenial
+LABEL maintainer="Shaun Jackman <sjackman@gmail.com>"
+
+RUN apt-get update \
+	&& apt-get install -y --no-install-recommends bzip2 ca-certificates curl file fonts-dejavu-core g++ git locales make openssh-client patch sudo uuid-runtime \
+	&& rm -rf /var/lib/apt/lists/*
+
+RUN localedef -i en_US -f UTF-8 en_US.UTF-8 \
+	&& useradd -m -s /bin/bash linuxbrew \
+	&& echo 'linuxbrew ALL=(ALL) NOPASSWD:ALL' >>/etc/sudoers
+ADD . /home/linuxbrew/.linuxbrew/Homebrew
+RUN cd /home/linuxbrew/.linuxbrew \
+	&& mkdir -p bin etc include lib opt sbin share var/homebrew/linked Cellar \
+	&& ln -s ../Homebrew/bin/brew /home/linuxbrew/.linuxbrew/bin/ \
+	&& chown -R linuxbrew: /home/linuxbrew/.linuxbrew \
+	&& cd /home/linuxbrew/.linuxbrew/Homebrew \
+	&& git remote set-url origin https://github.com/Homebrew/brew
+
+USER linuxbrew
+WORKDIR /home/linuxbrew
+ENV PATH=/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:$PATH \
+	SHELL=/bin/bash \
+	USER=linuxbrew
+
+# Install portable-ruby and tap homebrew/core.
+RUN HOMEBREW_NO_ANALYTICS=1 HOMEBREW_NO_AUTO_UPDATE=1 brew tap homebrew/core \
```

Do you want to install `git` as well to fix the `brew doctor` complaint?

```suggestion
RUN HOMEBREW_NO_ANALYTICS=1 HOMEBREW_NO_AUTO_UPDATE=1 brew tap homebrew/core \
	&& brew install git \
```

---

I guess that does bring in a few dependencies with it

---

I agree that ideally this would have no `brew doctor` complaints 👍

---

How about changing `check_git_version minimum_version` from 2.14.3 to 2.7.0? Ubuntu 16.04 LTS (Xenial Xerus) ships with git 2.7.4. That change could be for either macOS and Linux or only Linux, at your preference.

---

The only `brew doctor` complain is…
```
Warning: An outdated version (2.7.4) of Git was detected in your PATH.
Git 2.14.3 or newer is required for Homebrew.
```

---

I'd rather we kept the Git requirement as it is; we do actually use features that make it require the current version. For the Docker image a PPA would likely suffice.

---

Good idea!

---

https://launchpad.net/~git-core/+archive/ubuntu/ppa seems like a good candidate

---

Done!